llvm/docs/GitHub.rst

```diff
@@ -46,11 +73,106 @@ The default commit message for a squashed pull request is the pull request
 description, so this will allow reviewers to review the commit message before
 approving the commit.
 
+When pushing to your branch, make sure you push to the correct fork. Check your
+remotes with:
+
+::
+
+  git remote -v
+
+And make sure you push to the remote that's pointing to your fork.
+
+Rebasing Pull Requests and Force Pushes
+---------------------------------------
+In general, you should avoid rebasing a Pull Request and force pushing to the
+branch that's the root of the Pull Request during the review. This action will
+make the context of the old changes and comments harder to find and read.
```

Comment:
Isn't this too restrictive? Not rebasing your branch during review is too restrictive IMO. You might have conflicts with `main` at some point, so you have to rebase at some point to land it and it makes more sense for a reviewer to review the post-rebase changes in my opinion. 
Additionally, in such case I am assuming CI will fail as well, as it can only run if there are no conflicts with `main` right? Having to accept a review that is not tested by CI seems odd to me.

Would it make more sense to say "do not invalidate the commit history by squashing and ammeding commits", but that you can rebase and then force-push as needed? Or maybe this should be in the "Updating Pull Requests" section.

Reply:
Not putting much value in this (I think I agree more with you), but the closest thing to a consensus I could find was that people wanted to avoid rebases.

Reply:
I don't think this is a good advice here, unless you explicitly say: "you  should merge `origin/main` into your branch regularly".

Otherwise you may run into conflicts, or mid-air collision: any `ninja check` you run locally isn't testing against the latest `main` either.
It's not even clear what the pre-merge CI would test: would it be an outdated version of the code?

Reply:
I think *avoiding* rebases is a good policy (though I'd encourage folks to try my [diff-modulo-base](https://git.sr.ht/~nhaehnle/diff-modulo-base) tool to help in reviews), but it should be *allowed*.

That most closely reflects how things work with Phabricator.